Dev22

SleeplessShu opened the pull request:
Привет! Большое спасибо за проверку работы. Пожалуйста, если у тебя будет критика, то мне очень поможет, если помимо того как не надо делать, ты скажешь, как надо. Спасибо и хорошего дня!

SleeplessShu committed:
created playlist_editor.xml

SleeplessShu committed:
added playlist database
added mediateka/playlists tab.xml

SleeplessShu committed:
all important functions work correctly

SleeplessShu committed:
bottomNavigationView and view_line hides on player and playlist creation fragments

SleeplessShu committed:
fixed default image for albums without images

SleeplessShu committed:
added nice enter and exit animation of PlayerFragment

SleeplessShu committed:
added nice enter and exit animation between Mediateka, Settings and Search fragments

SleeplessShu committed:
part of code wich control bottom navigation visibility moved to separate function bottomNavigationVisible

SleeplessShu committed:
added enter and exit animation for playlist editor

SleeplessShu committed:
fixed double tap bug on button Like in Player

SleeplessShu committed:
fixed quality of the default image in Playlist Editor

SleeplessShu committed:
fixed quality of the default image in Playlist Editor

SleeplessShu committed:
fixed application crash when switching back from PlaylistEditor to Player

fixed player error on mediaplayerSetup when player is not prepared

SleeplessShu committed:
added permissions request on image selection in PlaylistFragment

SleeplessShu committed:
moved notification messages to strings resources

SleeplessShu committed:
fixed width of the text element in playlist unit

fixed layout of the elements in mediateka-playlists

SleeplessShu committed:
fixed bottom sheet offset and overlayVisibility

SleeplessShu committed:
fixed visibility of textInput fields in PlaylistEditor when using screenKeyboard

AleksandrKdev commented on app/src/main/java/com/practicum/playlistmaker_ver2/main/ui/MainActivity.kt:
```diff
@@ -18,14 +17,93 @@ class MainActivity : AppCompatActivity() {
+            } else {
+                bottomNavigationVisible(true)
+            }
+        }/*val bottomNavigationView = binding.bottomNavigationView
```
неиспользуемый код можно сразу удалять

SleeplessShu replied:
сделал

SleeplessShu replied:
удалил

AleksandrKdev commented on app/src/main/java/com/practicum/playlistmaker_ver2/main/ui/MainActivity.kt:
```diff
@@ -18,14 +17,93 @@ class MainActivity : AppCompatActivity() {
+
+        navController.addOnDestinationChangedListener { _, destination, _ ->
+            if (destination.id == R.id.playlistCreationFragment || destination.id == R.id.playerFragment) {
+                bottomNavigationVisible(false)
```
Можно сразу условие в метод передать

SleeplessShu replied:
передал

AleksandrKdev commented on app/src/main/java/com/practicum/playlistmaker_ver2/main/ui/MainActivity.kt:
```diff
@@ -18,14 +17,93 @@ class MainActivity : AppCompatActivity() {
+    }
 
+    private fun bottomNavigationVisible(isVisible: Boolean) {
+        if (isVisible) {
```
Тут можно экстеншены использовать .isVisible, .isGone

`findViewById<View>(R.id.bottomNavigationView)?.isVisible= isVisible`

SleeplessShu replied:
сделал, спасибо

AleksandrKdev commented on app/src/main/java/com/practicum/playlistmaker_ver2/player/data/repository/PlayerRepositoryImpl.kt:
```diff
@@ -1,18 +1,171 @@
+import com.practicum.playlistmaker_ver2.player.domain.repositories.PlayerRepository
 
+class PlayerRepositoryImpl(
+    private var player: MediaPlayer,
```
Тут вроде можно использовать val

AleksandrKdev commented on app/src/main/java/com/practicum/playlistmaker_ver2/player/data/repository/PlayerRepositoryImpl.kt:
```diff
@@ -1,18 +1,171 @@
+
+/*
+
+class PlayerRepositoryImpl(
```
Лишнее

SleeplessShu replied:
почистил

AleksandrKdev commented on app/src/main/java/com/practicum/playlistmaker_ver2/player/ui/backup_activityPlayer.kt:
```diff
@@ -0,0 +1,256 @@
+import com.practicum.playlistmaker_ver2.player.ui.models.PlayerTrack
+
+
+class ActivityPlayer : AppCompatActivity() {
```
Лишнее

SleeplessShu replied:
Удалил

AleksandrKdev commented on app/src/main/res/layout/playlist_editor.xml:
```diff
@@ -0,0 +1,125 @@
+            style="@style/CustomOutlineBox"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
```
marginHorizontal

AleksandrKdev replied:
16dp вынести в dimen

AleksandrKdev commented on app/src/main/res/layout/playlist_editor.xml:
```diff
@@ -0,0 +1,125 @@
+            style="@style/CustomOutlineBox"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
```
marginHorizontal

AleksandrKdev commented on app/src/main/res/layout/playlist_editor.xml:
```diff
@@ -0,0 +1,125 @@
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginHorizontal="@dimen/dim_16dp"
+            android:layout_marginTop="32dp"
```
marginVertical

AleksandrKdev replied:
32dp в dimen

SleeplessShu committed:
added name of playlist in notification message that track already in playlist

SleeplessShu committed:
added name of playlist in notification message that track successfully added in playlist

SleeplessShu committed:
added description.isEmpty check when user leave playlist editor without saving

SleeplessShu committed:
fixed behavior of the bottom navigation in search fragment keyboard hide it from user

SleeplessShu committed:
fixed behavior when screen orientation switching call critical error (removed arguments in constructor)

SleeplessShu committed:
fixed app crashing when call binding after onDestroyView()
fixed app crashing when switching the theme
fixed app crashing when change screenview or user switching to another app

SleeplessShu committed:
rearrange database folder between data and domain layers

SleeplessShu committed:
edits based on style advices

SleeplessShu committed:
edits based on style advices
some scales moved to dimens.xml
